### PR TITLE
CRM-16189 : Accrual accounting entries are not created correctly for fixed membership

### PR DESCRIPTION
**Pull request by monishdeb**
Overview
----------------------------------------
Fixes include:
1. Calculates and stores in DB the deferred revenue transactions details for each line item for all 3 types of contribution objects : memberships, registration, contributions.
2. It extends the FinancialTrxnText to ensure that deferred revenue db entries are properly created.
3. Corrects the calculation of membership revenue to recognize so that it now supports fixed membership periods properly.

---

 * [CRM-16189: Improve support for Accrual Method bookkeeping \(IIDA-102\)](https://issues.civicrm.org/jira/browse/CRM-16189)

### CODE REVIEW

**Comment by monishdeb** [comment]
This PR is created in favor of PR https://github.com/civicrm/civicrm-core/pull/8924 which was closed due to stale.

File: CRM/Contribute/BAO/Contribution.php [hunk 1]
```diff
@@ -3663,7 +3676,7 @@ public static function updateFinancialAccounts(&$params, $context = NULL) {
       }
     }
 
-    CRM_Core_BAO_FinancialTrxn::createDeferredTrxn(CRM_Utils_Array::value('line_item', $params), $params['contribution'], TRUE, $context);
+    CRM_Core_BAO_FinancialTrxn::createDeferredTrxn(CRM_Utils_Array::value('line_item', $params), $params['id'], TRUE, $context);
```
Comment (seamuslee001) [inline CRM/Contribute/BAO/Contribution.php]: @monishdeb tests are complaining about id being undefined here

Comment (monishdeb) [reply CRM/Contribute/BAO/Contribution.php]: Hmm, thanks for pointing that out .. will fix it, also this patch needs some additional work, so I marked with [WIP]

**Comment by eileenmcnaughton** [comment]
This is currently stale - I'm going to close & let you re-open when review-ready

**Pull request closed without merging**